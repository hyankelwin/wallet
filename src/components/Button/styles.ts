import styled from 'styled-components';

interface ButtonProps {
  background?: string;
  color?: string;
}

export const Content = styled.button<ButtonProps>`
  background-color: ${props => props.background};
  color: ${props => props.color};
  width: 100%;
  border: none;
  border-radius: 2px;
  padding: 16px;
  font-size: 16px;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.15);
  font-family: 'Montserrat';
  display: flex;
  justify-content: center;
  align-items: center;

  :hover {
    opacity: 0.8;
  }
`;
