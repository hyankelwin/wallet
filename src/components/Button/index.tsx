import React, { ButtonHTMLAttributes } from 'react';
import { Content } from './styles';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  background: string;
  color: string;
}

const Button: React.FC<ButtonProps> = ({ color, background, ...rest }) => {
  return (
    <Content type="button" color={color} background={background} {...rest} />
  );
};

export default Button;
