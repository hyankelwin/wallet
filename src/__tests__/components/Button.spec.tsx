import { mount } from 'enzyme';

import Button from '../../components/Button';
import Color from '../../styles/color';

describe('Button component', () => {
  it('should be able to render Button component with success', () => {
    const wrapper = mount(
      <Button color={Color.white} background={Color.primary.default} />,
    );

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
