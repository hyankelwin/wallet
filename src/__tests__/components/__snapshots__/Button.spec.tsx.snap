// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should be able to render Button component with success 1`] = `
"<Button color=\\"#FFFFFF\\" background=\\"#252a40\\">
  <styled.button type=\\"button\\" color=\\"#FFFFFF\\" background=\\"#252a40\\">
    <button type=\\"button\\" color=\\"#FFFFFF\\" className=\\"sc-bdvvtL kOqqac\\" />
  </styled.button>
</Button>"
`;
